### ISSUE TITLE
SelfSubjectAccessReview Fails via Port 443 on CNI Setup

### ISSUE THREAD

**Issue by steve-gray** [linkerd/linkerd2#10531]
### What is the issue?

Looks like the CNI doesn't like something about Oracle Linux when it comes to starting up in VCN native networking mode (i.e. it works *with* flannel, but not without)....

### How can it be reproduced?

- Set up a cluster using Oracle OKE using VCN native networking (disable flannel).
- Install the CNI
- Install Linkerd with CNI enabled``

### Logs, error output, etc

Identity pods fail to start, eventually timing out with:

```
kubectl logs -f -n linkerd linkerd-identity-678d49c968-4jwzz -c identity
time="2023-03-14T06:10:11Z" level=info msg="running version stable-2.12.3"

time="2023-03-14T06:10:41Z" level=fatal msg="Failed to initialize identity service: Post \"https://10.55.0.1:443/apis/authorization.k8s.io/v1/selfsubjectaccessreviews\": dial tcp 10.55.0.1:443: i/o timeout"
```

### output of `linkerd check -o short`

All failing/not running due to controlplane being totally KIA.

### Environment

- Kubernetes 1.25.4
- Oracle OKE
- Oracle Linux 8.7
- Linkerd 2.12.3 (down from .4 due to a pending hotfix for init container resources)

### Additional context

![image](https://user-images.githubusercontent.com/14563775/224912894-4f4b5367-4dab-406a-8e4b-8c6040814b53.png)

Attempting to install with 443 and 6443 (both, either) on the skip outbound ports list for the CNI and the control plane has no effect.

### Would you like to work on fixing this bug?

None

**Comment by steve-gray** [comment]
This is the same issue as:

- #7398 (Closed as wontfix)
- #7460 (Closed by @kleimkuhler due to a speculative fix in #9817 which isn't working here for the 2x ports)
- #6232 

I'm not sure if the work in #9841 will be a candidate for a fix here, but I'm in 2.12.3 as per @alpeb's comment in #7786 and not sure if I'm missing something.

Works fine the second I ditch the CNI.

**Comment by steve-gray** [comment]
I believe the issue equally affects nft-mode when working without the CNI. Here's the nft-save for the rules chain:

```
 Generated by iptables-nft-save v1.8.8 (nf_tables) on Tue Mar 14 07:45:50 2023
*nat
:PREROUTING ACCEPT [0:0]
:INPUT ACCEPT [0:0]
:OUTPUT ACCEPT [0:0]
:POSTROUTING ACCEPT [0:0]
:PROXY_INIT_OUTPUT - [0:0]
:PROXY_INIT_REDIRECT - [0:0]
-A PREROUTING -m comment --comment \"proxy-init/install-proxy-init-prerouting/1678779950\" -j PROXY_INIT_REDIRECT
-A OUTPUT -m comment --comment \"proxy-init/install-proxy-init-output/1678779950\" -j PROXY_INIT_OUTPUT
-A PROXY_INIT_OUTPUT -m owner --uid-owner 2102 -m comment --comment \"proxy-init/ignore-proxy-user-id/1678779950\" -j RETURN
-A PROXY_INIT_OUTPUT -o lo -m comment --comment \"proxy-init/ignore-loopback/1678779950\" -j RETURN
-A PROXY_INIT_OUTPUT -p tcp -m multiport --dports 443,6443 -m comment --comment \"proxy-init/ignore-port-443,6443/1678779950\" -j RETURN
-A PROXY_INIT_OUTPUT -p tcp -m comment --comment \"proxy-init/redirect-all-outgoing-to-proxy-port/1678779950\" -j REDIRECT --to-ports 4140
-A PROXY_INIT_REDIRECT -p tcp -m multiport --dports 4190,4191,4567,4568 -m comment --comment \"proxy-init/ignore-port-4190,4191,4567,4568/1678779950\" -j RETURN
-A PROXY_INIT_REDIRECT -p tcp -m comment --comment \"proxy-init/redirect-all-incoming-to-proxy-port/1678779950\" -j REDIRECT --to-ports 4143
COMMIT
# Completed on Tue Mar 14 07:45:50 2023
```

Pods on same host without the proxy loaded are able to hit the kube API on 10.55.0.1 fine.
